### ISSUE TITLE
Savepoint error logged when org is created

### ISSUE THREAD

**Issue by BruceMacD** [infrahq/infra#2904]
## Describe the bug
When the Infra server starts the following error can sometimes be seen:
```
{"level":"error","rows":0,"query":"SAVEPOINT beforeCreate","elapsed":0.925375,"error":"ERROR: SAVEPOINT can only be used in transaction blocks (SQLSTATE 25P01)","time":1660162230114,"caller":"postgres@v1.3.7/postgres.go:201"}
```

The server will start normally after this.

## Environment

PostgreSQL 14.4

## Relevant Infra Logs

```
{"level":"error","rows":0,"query":"SAVEPOINT beforeCreate","elapsed":0.925375,"error":"ERROR: SAVEPOINT can only be used in transaction blocks (SQLSTATE 25P01)","time":1660162230114,"caller":"postgres@v1.3.7/postgres.go:201"}
```

**Comment by ssoroka** [comment]
this happens every time the org create db call is hit

**Comment by dnephin** [comment]
I think it's not specific to orgs. It happens with any call to `data.add` that is outside a transaction, which is the case for all data tests  right now.

**Comment by dnephin** [comment]
I think one nice way of solving this would be to remove the need for savepoints. I think if we use `ON CONFLICT DO NOTHING` (https://www.postgresql.org/docs/current/sql-insert.html#SQL-ON-CONFLICT) we can remove the savepoint that we added for create grants.

**Comment by BruceMacD** [comment]
This specific case was resolved in f45166a
